src/main/java/bio/terra/stairway/FlightDao.java

```diff
@@ -1,301 +1,371 @@
+        }
     }
 
-    @Transactional(propagation = Propagation.REQUIRED)
-    public void delete(String flightId) {
+    public void delete(String flightId) throws DatabaseOperationException {
         final String sqlDeleteFlightLog = "DELETE FROM " + FLIGHT_LOG_TABLE + " WHERE flightid = :flightid";
         final String sqlDeleteFlight = "DELETE FROM " + FLIGHT_TABLE + " WHERE flightid = :flightid";
 
-        MapSqlParameterSource params = new MapSqlParameterSource()
-            .addValue("flightid", flightId);
+        try (Connection connection = dataSource.getConnection();
+             NamedParameterPreparedStatement deleteFlightStatement =
+                 new NamedParameterPreparedStatement(connection, sqlDeleteFlight);
+             NamedParameterPreparedStatement deleteLogStatement =
+                 new NamedParameterPreparedStatement(connection, sqlDeleteFlightLog)) {
+
+            connection.setAutoCommit(false);
 
-        jdbcTemplate.update(sqlDeleteFlightLog, params);
-        jdbcTemplate.update(sqlDeleteFlight, params);
+            deleteFlightStatement.setString("flightid", flightId);
+            deleteFlightStatement.getPreparedStatement().executeUpdate();
+
+            deleteLogStatement.setString("flightid", flightId);
+            deleteLogStatement.getPreparedStatement().executeUpdate();
+
+            connection.commit();
+
+        } catch (SQLException ex) {
+            throw new DatabaseOperationException("Failed to delete flight", ex);
+        }
     }
 
-    public boolean ownsFlight(String flightId, String subject) {
-        final String sqlFlight = "SELECT owner_id" +
+    boolean ownsFlight(String flightId, String subject) throws DatabaseOperationException {
+        final String sqlFlight = "SELECT COUNT(*) AS matches" +
             " FROM " + FLIGHT_TABLE +
-            " WHERE flightId = :flightid";
-        MapSqlParameterSource params = new MapSqlParameterSource()
-            .addValue("flightid", flightId);
-        String flightSubject = jdbcTemplate.queryForObject(sqlFlight, params, String.class);
-        return subject.equals(flightSubject);
+            " WHERE flightId = :flightid" +
+            " AND owner_id = :ownerid";
+
+        long matches = 0;
+
+        try (Connection connection = dataSource.getConnection();
+             NamedParameterPreparedStatement ownsFlight =
+                 new NamedParameterPreparedStatement(connection, sqlFlight)) {
+
+            ownsFlight.setString("flightid", flightId);
+            ownsFlight.setString("ownerid", subject);
+
+            try (ResultSet rs = ownsFlight.getPreparedStatement().executeQuery()) {
+                while (rs.next()) {
+                    matches = rs.getLong("matches");
+                }
+            }
+        } catch (SQLException ex) {
+            throw new DatabaseOperationException("Failed to get flight list", ex);
+        }
+
+        return (matches == 1);
```

Comment:
flightId is the primary key, so this ResultSet should only ever have return one row with matches=0 or 1 right? If so, then a small clarity nitpick: I think it would be clearer if you used either SQL COUNT or counted the rows in the while loop in Java. For the latter case, maybe you could remove the while loop, just call next() once and then add a comment that matches is 0 or 1 (or thrown an exception if it's neither).

Reply:
Good catch. Don't know what I was thinking! I'll remove the while loop and add a comment about why we know the count is 0 or 1.